perception/traffic_light_arbiter/src/traffic_light_arbiter.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright 2023 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <lanelet2_extension/utility/message_conversion.hpp>
+#include <rclcpp/time.hpp>
+#include <traffic_light_arbiter/traffic_light_arbiter.hpp>
+
+#include <lanelet2_core/primitives/BasicRegulatoryElements.h>
+
+#include <map>
+#include <memory>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace lanelet
+{
+
+using TrafficLightConstPtr = std::shared_ptr<const TrafficLight>;
+
+std::vector<TrafficLightConstPtr> filter_traffic_signals(const LaneletMapConstPtr map)
+{
+  std::vector<TrafficLightConstPtr> signals;
+  for (const auto & element : map->regulatoryElementLayer) {
+    const auto signal = std::dynamic_pointer_cast<const TrafficLight>(element);
+    if (signal) {
+      signals.push_back(signal);
+    }
+  }
+  return signals;
+}
+
+}  // namespace lanelet
+
+TrafficLightArbiter::TrafficLightArbiter(const rclcpp::NodeOptions & options)
+: Node("traffic_light_selector", options)
+{
+  external_time_tolerance_ = this->declare_parameter<double>("external_time_tolerance", 5.0);
+  perception_time_tolerance_ = this->declare_parameter<double>("perception_time_tolerance", 1.0);
```

Comment:
minor: these parameters seem to be different from the readme. Is it ok?

Reply:
Addressed in [53b3540](https://github.com/autowarefoundation/autoware.universe/pull/3780/commits/53b35404ea1764c4f963bdd57e65ce7fc1db291f).
:bow:

Reply:
@knzo25 LGTM!